src/ESM.sol

```diff
@@ -26,25 +26,32 @@ interface GemLike {
 }
 
 interface EndLike {
+    function live() external view returns (uint256);
+    function vat()  external view returns (address);
     function cage() external;
 }
 
+interface VatLike {
+    function deny(address) external;
+}
+
 contract ESM {
-    GemLike public immutable gem; // collateral
-    EndLike public immutable end; // cage module
-    address public immutable pit; // burner
-    uint256 public immutable min; // threshold
-    uint256 public fired;
+    GemLike public immutable gem;   // collateral
+    EndLike public immutable end;   // cage module
+    address public immutable proxy; // Pause proxy
+    address public immutable pit;   // burner
```

Comment:
I don't really understand why we have `pit`. Why not just hardcode a transfer to `address(0)` or `address(this)`? It seems like unnecessary complexity.

Reply:
To actually have a real `burn` of the MKR delegated to another contract (in the case we want to keep this same MKR token after shutdown).
At that point there wasn't `mkr-authority` which allows now everybody to `burn` their MKR.
Probably the easier is just to do `mkr.burn(amt)` now that we have the authority, but that makes this module dependant on the authority still assigned to the MKR token to work.

Reply:
Yeah that's why I didn't suggest to call `burn` in this PR--on the off chance that for some reason the authorization isn't there for the ESM, that would be quite bad as the ESM wouldn't work at all.

For burning--an easy way to add it, assuming we transfer tokens to `address(this)` in `join`, would be to add another function on the ESM that `burns` all the MKR it holds when called. Then burning is off the critical path, and we wouldn't need `pit`.

Reply:
Sounds a good solution to me.

Reply:
Done!